server/src/internalClusterTest/java/org/elasticsearch/snapshots/FeatureStateResetApiIT.java

```diff
@@ -94,6 +101,25 @@ public void testResetSystemIndices() throws Exception {
         assertThat(response.getIndices(), arrayContaining("my_index"));
     }
 
+    /**
+     * Evil test - what if the cleanup method fails?
+     */
+    public void testFailFastOnStateCleanupFailure() throws Exception {
```

Comment:
technically this is not failing fast since we let everything finish before returning a response?

Reply:
Good catch — I've changed the name.